runtime/storage_test.go

```diff
@@ -2268,3 +2268,517 @@ transaction {
 	)
 	require.NoError(t, err)
 }
+
+func TestRuntimeReferenceOwnerAccess(t *testing.T) {
```

Comment:
Just thinking out loud here: What would be the (expected) behavior if:
- A resource value `x` is created
- A reference `xRef` is created for `x`
- `x` is stored in account `A`
- load the value in account `A`; say `y`
- Take a reference `yRef` to `y`
- Store `y` in a different account `B`
-  What would be the owners of `xRef` and `yRef` respectively 🤔 

Not sure, maybe worth adding a test-case?

Reply:
After this operation, both references point to the same resource, which is stored in account B, so they should both report the owner to be B.

I wrote a test for this, but it is not passing, I'm looking into it.

Thank you for coming up with another test case!

Reply:
The tracking had a bug where it would loose the tracked resources when the source and target ID where the same, which was only exercised through your test case (!). Thank you again for encouraging the "basic" case (resource is not in a container or nested), but also testing the behaviour over multiple operations 👌 

Fixed in 1dc3d579a7b14886fe370d13c9d50d29530ac0f2

Reply:
Nice! Thank you for adding the test and for the fix! 👍